meerk40t/core/cutplan.py

```diff
@@ -639,6 +643,88 @@ def clear(self):
         self.plan.clear()
         self.commands.clear()
 
+    def optimize_rasters(self, operation_list, op_type):
+        def generate_clusters(operation):
+            clusters = list()
+            cluster_bounds = list()
+            for node in operation.children:
+                if node.type == "reference":
+                    node = node.node
+                try:
+                    bb = node.paint_bounds
+                except AttributeError:
+                    continue
+                clusters.append([node])
+                cluster_bounds.append((bb[0], bb[1], bb[2], bb[3]))
+
+            needs_repeat = True
+            while needs_repeat:
```

Comment:
A very very minor point, but would a `while true:` and use of `break` be a better code style?

Reply:
He's breaking the inner loop there at a particular point. So it's a lot harder to break correctly, and no. You'd use a function for this for better code style. Nearly always when you're using `while true` and `break` you're better off writing the inner loop as a function and returning.

Though you might be able to do multiple merges if you iterated backwards.

Reply:
> He's breaking the inner loop there at a particular point. So it's a lot harder to break correctly, and no. You'd use a function for this for better code style. Nearly always when you're using `while true` and `break` you're better off writing the inner loop as a function and returning.
> 
> Though you might be able to do multiple merges if you iterated backwards.

Use of a function - I agree. This comment was (as stated) a very minor one.

Multiple merges - I think the code has removed the entry you are comparing with so you need to break and move to the next one down the list.